compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.h

```diff
@@ -19,14 +19,109 @@ namespace iree_compiler {
 // firstly. At this stage, we do not apply vectorization. The reduction dim
 // won't get tiled if the case is matmul + generic op. In this case, we have to
 // tile along reduction dim again, which needs them to be TilingInterface ops.
-enum class TilingLevel : unsigned {
-  // Tile TilingInterface operations to threads.
-  WorkGroupTiles = 0,
-  // Tile TilingInterface operation on workgroup thread for parallel dims.
-  ParallelTiles = 1,
-  // Tile TilingInterface operations on workgroup thread for reduction dims.
-  ReductionTiles = 2,
-  NumTileLevels = 3
+// TODO: Update doc.
+class TilingConfig {
+ public:
+  TilingConfig(IREE::Codegen::LoweringConfigAttr lc) : loweringConfig(lc) {
+    int numTileLevels = loweringConfig.getTileSizes().size();
+
+    // Initialize indices as invalid.
+    int invalidIdx = std::numeric_limits<unsigned>::max();
```

Comment:
I think this should be fixed/constant value and the type should be `unsigned`. let's use `constexpr unsigned kInvalidIdx`?

Or maybe we should just declare the invalidIdx in the enum class..